### DISCUSSION TITLE
Is UFS able to output dynf*.nc and phyf*.nc in NETCDF4 format?

### DISCUSSION THREAD

**Discussion by lo-y-wni** [ufs-community/ufs-srweather-app#795] [Q&A] [answered]
Hello everyone:

Since UFS outputs are heavy, so I want to know does it currently support NETCDF-4 format to decrease the data size?

Bests,
Yi-Chuan

**Comment by EdwardSnyder-NOAA** [comment]
Good afternoon Yi-Chuan,

The [documentation](https://ufs-srweather-app.readthedocs.io/en/develop/InputOutputFiles.html#id3) doesn't specify which format of netCDF the UFS WM is using, so I did a ncdump on the `dynf*.nc` and `phyf*.nc` files and can confirm that they are 'netCDF-4 classic model' format. 

Please let us know if you have any more questions!

Thanks!

Edward Snyder | Advanced User Support Team

**Reply by lo-y-wni** [reply]
Hi Edward:

As I know netCDF-4 classic model doesn't do any data compression, so the size is large. And I hope to switch the format to netCDF-4 which can reduce file size with compression.
I found the there is a section in namelist called mpp_io_nml which contain deflate_level and shuffle. It is used in FV3_GFS_v16 suites. Are these options about data compression?

Bests,
Yi-Chuan

**Reply by EdwardSnyder-NOAA** [reply, marked as answer]
Hi Yi-Chuan,

My apologies for the delayed response. To compress the `dynf*.nc` and `phyf*.nc` files, go into `ufs-srweather-app/parm/model_configure` and modify the `ideflate` variable before running the `run_fcst_mem###` step of your experiment. The accepted values for this variable are 0-9 as mentioned [here](https://ufs-weather-model.readthedocs.io/en/ufs-v1.0.0/InputsOutputs.html#model-configure-file). 

The `mpp_io_nml` section of the FV3.input.yml refers to a package within the [FMS framework](https://github.com/NOAA-GFDL/FMS/blob/main/mpp/mpp_io.F90) that is designed to aid with reading and writing of files in a parallel environment.

Thanks!

Edward Snyder | Advanced User Support Team

**Reply by lo-y-wni** [reply]
Hi Edward,

Many thanks your response. It's a very useful information. I'll try `ideflate` later to see it's affect on data size.
One more question is about the [FMS framework](https://github.com/NOAA-GFDL/FMS/blob/main/mpp/mpp_io.F90). I'm curious that when should I use this function? Because only GFS suite uses it, it seems that HRRR and RAP doesn't turn it on. If I set `mpp_io_nml` properly, I can reduce the I/O elapsed time?

Best,
Yi-Chuan

**Reply by EdwardSnyder-NOAA** [reply]
Hi Yi-Chuan,

After some further digging, it appears that the `mpp_io_nml` block of code is [currently broken](https://github.com/ufs-community/ufs-weather-model/issues/1500) for atmosphere but works fine with the ocean model. In addition, this piece of code is to compress the restart netCDF files. In the SRW App, they are found under your `<experiment name>/<experiment date>/RESTART`. There appears to be a fix in the works to address this bug. Depending on what [deflate_level](http://climate-cms.wikis.unsw.edu.au/NetCDF_Compression_Tools#General_guidelines) you set could help reduce the I/O elapsed time when writing the restart netCDF files.

Please let us know if you have any additional questions!

Thanks,

Edward Snyder | Advanced User Support Team
